sdk/trace/provider.go

```diff
@@ -99,6 +102,14 @@ func NewTracerProvider(opts ...TracerProviderOption) *TracerProvider {
 	return tp
 }
 
+// SetClock supports setting a mock clock for testing or some clock with custom
+// time correction logic.
+func (p *TracerProvider) SetClock(clock tracetime.Clock) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	p.clock = clock
+}
```

Comment:
This seems more in line with the other configuration options we have.  What advantages do we get with this form over having it configured via a `TraceProviderOption`? Like `WithBatcher`

Reply:
Agreed, an option like `WithClock` would be more coherent. Will make the change.